src/main/kotlin/net/maizegenetics/phgv2/cli/ListSamples.kt

```diff
@@ -0,0 +1,135 @@
+package net.maizegenetics.phgv2.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.enum
+import net.maizegenetics.phgv2.utils.getBufferedWriter
+import net.maizegenetics.phgv2.utils.retrieveAgcData
+import net.maizegenetics.phgv2.utils.verifyURI
+import org.apache.logging.log4j.LogManager
+import kotlin.test.DefaultAsserter.fail
+
+/**
+ * This class is used to list the sample names in both the AGC compressed file and the tiledb datasets.
+ * It is a command that allows users to see what data they have loaded, and make decisions on
+ * which sample data to include when imputing, analysing or exporting data.
+ *
+ * It takes as input a path to the tiledb datasets, a dataset type for output,  and an output file.
+ * By default, the output will be a file, one sample per line, showing
+ * the sample names that exist in the hvcf dataset.
+ *
+ * The optional parameter --data-set, allows the user to specify the dataset from which to pull sample names.  The default is hvcf.
+ * The options are agc, gvcf, hvcf, and all.  If all is selected, the samples from all datasets are listed in a tab-delimited file
+ * with columns for the sample name, and whether the sample is in the AGC file, the gvcf dataset and/or the hvcf dataset.
+ *
+ * If "hvcf", "gvcf" or "agc" is selected, the output will be a file with a list of sample names in the selected dataset, one per line
+ *
+ * Another optional parameter is the conda environment prefix.  If provided, this should be the full path to the conda environment.
+ *
+ */
+
+// Create an enum class to hold the dataset options
+enum class DatasetOptions {
+    agc, gvcf, hvcf, all
+}
+class ListSamples: CliktCommand(help = "List the sample names in both the AGC compressed file and the tiledb datasets.") {
+    private val myLogger = LogManager.getLogger(ListSamples::class.java)
+
+    val dbPath by option(help = "Folder name where TileDB datasets and AGC record is stored.  If not provided, the current working directory is used")
+        .default("")
+
+    // Datset option must be one from the enum, default to hvcf
+    val dataSet by option(help = "Name of the dataset from which to list the samples. Options are: all, agc, gvcf, hvcf.")
+        .enum<DatasetOptions>()
+        .default(DatasetOptions.hvcf)
```

Comment:
Not sure but I think Zack wanted "all" to be the default

Reply:
I thought @zrm22  agreed hvcf was the most common usage and should be the default.  It is easy to change.